avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -252,4 +231,89 @@ private Schema getReaderSchema(Schema writerSchema) {
     }
     return readerSchema;
   }
+
+  class DeserializationContext {
+    private final String topic;
+    private final Boolean isKey;
+    private final ByteBuffer buffer;
+    private final int schemaId;
+
+    private Schema cachedSchemaFromRegistry = null;
```

Comment:
I was trying to understand if there were any benefits to this cached instances. The CachedSchemaRegistryClient already caches the schema. Unless there are specific reasons, I would prefer to not cache here and simplify the code.